nntrainer/tensor/tensor_base.cpp

```diff
@@ -40,13 +40,28 @@ bool TensorBase::operator==(const TensorBase &rhs) const {
   return true;
 }
 
+void TensorBase::setTensorVar(TensorDim d, void *buf, size_t offset) {
+  dim = d;
+  strides = d.computeStrides();
+  /// Tensor does not own the memory
+  data = std::shared_ptr<MemoryData>(new MemoryData((void *)buf),
+                                     std::default_delete<MemoryData>());
+  offset = offset;
+}
+
```

Comment:
this is simply a helper function to set certain TensorBase variables in the Tensor class.
since Map() is a static function, the possible way to set TensorBase variables would be using a Tensor public function. 
therefore, the SetTensorVar() function has been newly added.